framework/src/variables/MooseVariableFE.C

```diff
@@ -831,6 +834,432 @@ MooseVariableFE<OutputType>::clearAllDofIndices()
   _lower_data->clearDofIndices();
 }
 
+template <typename OutputType>
+typename MooseVariableFE<OutputType>::ValueType
+MooseVariableFE<OutputType>::evaluate(const NodeArg & node_arg, const StateArg & state) const
+{
+  mooseAssert(node_arg.node, "Must have a node");
+  mooseAssert(this->hasBlocks(node_arg.subdomain_id),
+              "Our variable should be defined on the requested subdomain ID");
+  const Node & node = *node_arg.node;
+  mooseAssert(node.n_dofs(this->_sys.number(), this->number()),
+              "Our variable must have dofs on the requested node");
```

Comment:
we could also assert that this dof be a value dof as opposed to a gradient dof

and/or that there be only one dof on the node (per component if vector)

Reply:
I don't know what a gradient dof is. I don't know of finite element approximate formulations other than u = sum(i) phi_i * u_i

Reply:
I think it s a thing in our exotic families, like szabab. This is the kind of thing I just ask Roy about tbh

We essentially do not support them in MOOSE in both the auxkernel and transfer systems